### ISSUE TITLE
[FEA] Reduce time required to import cudf_polars

### ISSUE THREAD

**Issue by beckernick** [rapidsai/cudf#16244]
As noted in https://github.com/pola-rs/polars/pull/17550, import times are initially going to be a little slow but should improve over the next few months.

> The import of cudf_polars is currently quite expensive (will improve in the coming months), so since we lazy-load the gpu engine, the first query executed on gpu will pay this (one-time) cost.

(Tracking issue in the context of the Polars backend)

**Comment by wence-** [comment]
> For posterity, to profile import time, run
> 
> ```
> python -X importtime -c "import cudf_polars"
> ```
> 
> As of right now, this gives something a really long output (so I'm not going to paste it) since cudf imports a lot of stuff (e.g. pandas and co. which probably add to the long import time)

You can see a nice interactive flamegraph in the browser of the time if you do:
```
pip install tuna
python -X importtime -c "import cudf_polars" 2> import.log
tuna import.log
```

**Comment by wence-** [comment]
This will largely be addressed in 24.10 where pylibcudf is a separate package:

On current nightlies:

- `time python -c "import cudf"` -> ~1.5s (this is a proxy for what we currently have)
- `time python -c "import pylibcudf"` -> ~0.2s (this is a proxy for what we will have)

**Comment by wence-** [comment]
In 24.10, once polars is imported, importing cudf_polars now takes 0.2s, so I'm going to call this done.
